include/cpptoml.h

```diff
@@ -274,16 +288,18 @@ struct value_traits<T, true>
 template <class T>
 class value : public base
 {
+    friend std::shared_ptr<typename value_traits<T>::type>
+        cpptoml::make_value<>(T&& val);
+
+    friend std::shared_ptr<typename value_traits<T>::type>
+        cpptoml::make_value<>(T& val);
+
+    friend std::shared_ptr<typename value_traits<T>::type>
+        cpptoml::make_value<>(const T& val);
```

Comment:
So this is good except for the case where we want to allow things like

```cpp
table->insert("hello", "world"); // this worked before because of value_traits
                                 // even though const char* is not a valid_value,
                                 // it can be converted to one
```

Since I'm not too concerned with Machiavellian people, we can be generous and just friend the whole class of functions with

```cpp
template <class U>
friend std::shared_ptr<typename value_traits<U>::type> make_value(U&& val);
```